Sources/Basic/TerminalController.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+import func POSIX.getenv
+
+/// Tries to get the terminal width first using COLUMNS env variable and
+/// if that fails ioctl method testing on stdout stream.
+///
+/// - Returns: Current width of terminal if it was determinable.
+public func terminalWidth() -> Int? {
```

Comment:
I would make this a static function on `TerminalController` to limit the API surface from Basic.